### PR TITLE
Commando18 commerce 4866

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
Added e method to update ERC of a product

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
change requested here https://github.com/liferay-commerce/liferay-portal/pull/440 applied

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionServiceImpl.java [hunk 1]
```diff
@@ -328,6 +328,42 @@ public CPDefinition updateCPDefinition(
...
+			int expirationDateHour, int expirationDateMinute,
+			boolean neverExpire, String externalReferenceCode,
+			ServiceContext serviceContext)
+		throws PortalException {
+
+		_checkCommerceCatalogByCPDefinitionId(
+			cpDefinitionId, ActionKeys.UPDATE);
+
+		CPDefinition cpDefinition =
+			cpDefinitionLocalService.getCPDefinition(cpDefinitionId);
```
Comment (dsitu) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionServiceImpl.java:352]: I think this should satisfy SF.

```suggestion
		CPDefinition cpDefinition = cpDefinitionLocalService.getCPDefinition(
			cpDefinitionId);
```

**Review by dsitu — CHANGES_REQUESTED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
thanks @dsitu

**Review by dsitu — CHANGES_REQUESTED** [review]
@andrea-ale-sbarra thanks for making the SF changes! Can you separate the regen code into its own commit for the SF commit? thanks!

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionServiceImpl.java [hunk 2]
```diff
@@ -377,6 +377,23 @@ public void updateCPDisplayLayout(
 			CPDefinition.class, cpDefinitionId, layoutUuid, serviceContext);
 	}
 
+	@Override
+	public CPDefinition updateCProductExternalReferenceCode(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionServiceImpl.java:381]: update cp definition

**Comment by andrea-ale-sbarra** [comment]
add integration test

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:forward

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Pull request closed without merging**